lib/srv/sess.go

```diff
@@ -1271,6 +1271,9 @@ func sessionsStreamingUploadDir(ctx *ServerContext) string {
 }
 
 func (s *session) broadcastResult(r ExecResult) {
+	s.mu.Lock()
```

Comment:
I think this can deadlock.

`(*SessionRegistry).broadcastResult()` acquires the lock and then calls `(*session).boradcastResult()`, which will attempt to acquire a lock that it already holds.

Reply:
@zmb3 I don't think this is the case. The fist lock as you pointed out is in `SessionRegistry` and the one that I've added is in `session`
https://github.com/gravitational/teleport/blob/7c7bb75f22e158065e861583ce13e3480e511939/lib/srv/sess.go#L67-L68
https://github.com/gravitational/teleport/blob/7c7bb75f22e158065e861583ce13e3480e511939/lib/srv/sess.go#L530-L531
Alternative solution would be to move the `SessionRegistry` lock inside `(*SessionRegistry).findSessionLocked()` as I don't think that is needed during `(*session).broadcastResult()` call and leave the extra lock that I've added, but I'm not familiar with this part of code.